golangci-config/.golangci.yml

```diff
@@ -17,6 +17,7 @@ linters:
     - bodyclose
     - deadcode
     - errcheck
+    - gocyclo
```

Comment:
**suggestion:** I am using [gocognit](https://github.com/uudashr/gocognit) instead of `gocyclo`, I think it takes more things into consideration. It is configured the same way e.g. :
```
linters-settings:
  gocognit:
    min-complexity: 15
```
[Here is the whitepaper](https://www.sonarsource.com/docs/CognitiveComplexity.pdf) regarding cognitive complexity.

Reply:
thanks @MiroslavGatsanoga 

in very unscientific way I ran gocyclo on several random projects.
When looking at top 10 "offenders" I saw cyclomatic complexity rarely going above 11 or 12, thats why I decided that 15 is good value. Not a lot of changes on our side and yet useful.

Running gocongit on the same projects reports almost twice as high value for cognitive complexity.
Do you suggest using 15 (or it is just example) ? If not, what kind of value would you recommend.

Reply:
It was an example, but I think it might be used actually. Locally I have set 12 as `min-complexity` which I think is too conservative (even I ignore it sometimes), but I like to see if there are some functions that are borderline complex just to give a quick thought on how to refactor them in the future. 
Given the fact that linter failures will need to be addressed by everyone I think we can use the more liberal values, anything in the 15-20 range should be ok as a starting value, that could be brought down in the future.